apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.Statistics.cs

```diff
@@ -25,37 +26,46 @@ public partial class SessionPool
+            public int TotalReadWritePoolCount => PerSessionPoolSegmentStatistics.Sum(d => d.ReadWritePoolCount);
 
             /// <summary>
-            /// The total of <see cref="DatabaseStatistics.ActiveSessionCount"/> values across all databases in the pool.
+            /// The total of <see cref="SessionPoolSegmentStatistics.ActiveSessionCount"/> values across all databases in the pool.
             /// </summary>
-            public int TotalActiveSessionCount => PerDatabaseStatistics.Sum(d => d.ActiveSessionCount);
+            public int TotalActiveSessionCount => PerSessionPoolSegmentStatistics.Sum(d => d.ActiveSessionCount);
 
             /// <summary>
-            /// The total of <see cref="DatabaseStatistics.InFlightCreationCount"/> values across all databases in the pool.
+            /// The total of <see cref="SessionPoolSegmentStatistics.InFlightCreationCount"/> values across all databases in the pool.
             /// </summary>
-            public int TotalInFlightCreationCount => PerDatabaseStatistics.Sum(d => d.InFlightCreationCount);
+            public int TotalInFlightCreationCount => PerSessionPoolSegmentStatistics.Sum(d => d.InFlightCreationCount);
 
             /// <summary>
-            /// The total of <see cref="DatabaseStatistics.PendingAcquisitionCount"/> values across all databases in the pool.
+            /// The total of <see cref="SessionPoolSegmentStatistics.PendingAcquisitionCount"/> values across all databases in the pool.
             /// </summary>
-            public int TotalPendingAcquisitionCount => PerDatabaseStatistics.Sum(d => d.PendingAcquisitionCount);
+            public int TotalPendingAcquisitionCount => PerSessionPoolSegmentStatistics.Sum(d => d.PendingAcquisitionCount);
 
             /// <summary>
-            /// The statistics broken down by database. This is never null, but may be empty.
+            /// The statistics broken down by <see cref="SessionPoolSegmentKey"/>. This is never null, but may be empty.
             /// </summary>
-            public IReadOnlyList<DatabaseStatistics> PerDatabaseStatistics { get; }
+            [Obsolete("Use PerSessionPoolKeyStatistics instead. Both properties return the same data, but PerSessionPoolKeyStatistics name" +
+                " better represents the fact that sessions are pooled based on aspects other than database name.")]
+            public IReadOnlyList<DatabaseStatistics> PerDatabaseStatistics => PerSessionPoolSegmentStatistics.Select(s => new DatabaseStatistics(s)).ToList().AsReadOnly();
 
-            internal Statistics(IReadOnlyList<DatabaseStatistics> perDatabaseStatistics) =>
-                PerDatabaseStatistics = perDatabaseStatistics;
+            /// <summary>
+            /// The statistics broken down by database. This is never null, but may be empty.
+            ///</summary>
+            public IReadOnlyList<SessionPoolSegmentStatistics> PerSessionPoolSegmentStatistics { get; }
```

Comment:
I think this could just be PerSegmentStatistics (corresponding to the previous PerDatabaseStatistics). We're already in SessionPool.Statistics; I think the session-pool-ness is implicit.

Reply:
+1